src/main/java/org/mule/extension/api/introspection/ParameterModel.java

```diff
@@ -45,6 +45,14 @@
     boolean isRequired();
 
     /**
+     * Whether or not this parameter should be evaluated or treated as a literal
+     * value instead.
+     *
+     * @return a boolean value saying if the parameter is literal or not
+     */
+    boolean isLiteral();
```

Comment:
not a good idea to do this. You start here but will end up adding getters on the ParameterModel per each thing we add on any collaborator. The ParameterModel already has a getExpressionSupport() and that's all that's needed